File: app/discover/scan_environment.rb
#!/usr/bin/env ruby

# scan a environment for projects and regions

require 'singleton'
require_relative 'db_fetch_regions'
require_relative 'api_fetch_projects'
require_relative 'api_fetch_regions'
require_relative 'scan_project'

class ScanEnvironment < Scanner
  include Singleton
  
  def initialize()
    super([
      {
        :type => "project",
        :fetcher => ApiFetchProjects.new(),
        :object_id_to_use_in_child => "name",
        :children_scanner => ScanProject.instance
      }
    ])
  end
  
end

File: app/discover/scan_environment.py
#!/usr/bin/env python3
# scan an environment for projects

from singleton import Singleton
from scanner import Scanner
from folder_fetcher import FolderFetcher

class ScanEnvironment(Scanner, metaclass=Singleton):
  
  def __init__(self):
    super(ScanEnvironment, self).__init__([
      {
        "type": "regions_folder",
        "fetcher": FolderFetcher("regions", "environment"),
        "children_scanner": "ScanRegionsRoot"
      },
      {
        "type": "projects_folder",
        "fetcher": FolderFetcher("projects", "environment"),
        "children_scanner": "ScanProjectsRoot"
      }
    ])
